mdn/src/test/scala/org/apache/james/mdn/MDNReportParserTest.scala

```diff
@@ -0,0 +1,202 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *    http://www.apache.org/licenses/LICENSE-2.0                *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mdn
+
+import org.apache.james.mdn.`type`.DispositionType
+import org.apache.james.mdn.action.mode.DispositionActionMode
+import org.apache.james.mdn.fields.{AddressType, Disposition, Error, ExtensionField, FinalRecipient, Gateway, OriginalMessageId, OriginalRecipient, ReportingUserAgent, Text}
+import org.apache.james.mdn.modifier.DispositionModifier
+import org.apache.james.mdn.sending.mode.DispositionSendingMode
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class MDNReportParserTest {
+
+  @Test
+  def parseShouldReturnEmptyWhenMissingFinalRecipient(): Unit = {
+    val missing = "Disposition: automatic-action/MDN-sent-automatically;processed\r\n"
+    val actual = MDNReportParser.parse(missing).toOption
+    assertThat(actual.isEmpty)
```

Comment:
can we check that failed is defined here or in a similar test?

Reply:
What do you mean ? 
An Option is either empty or defined. It is not a Schrödinger Option :D